src/PhpSpec/Locator/BestMatchResourceManager.php

```diff
@@ -0,0 +1,123 @@
+<?php
+
+/*
+ * This file is part of PhpSpec, A php toolset to drive emergent
+ * design by specification.
+ *
+ * (c) Marcello Duarte <marcello.duarte@gmail.com>
+ * (c) Konstantin Kudryashov <ever.zet@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace PhpSpec\Locator;
+
+use PhpSpec\Exception\Locator\ResourceCreationException;
+
+final class BestMatchResourceManager implements ResourceManager
+{
+    /**
+     * @var ResourceLocator[]
+     */
+    private $locators = array();
+
+    /**
+     * @param ResourceLocator $locator
+     */
+    public function registerLocator(ResourceLocator $locator)
+    {
+        $this->locators[] = $locator;
+
+        @usort($this->locators, function (ResourceLocator $locator1, ResourceLocator $locator2) {
```

Comment:
There's some historical issue that needs it - I think PHP believes the array has changed while being sorted and issues a warning?